### ISSUE TITLE
Make no-useless-catch a core rule

### ISSUE THREAD

**Issue by agrasley** [eslint/eslint#11174]
**Please describe what the rule should do:**

Prevents useless `catch` clauses that do nothing but rethrow the caught error. This rule already exists as an ESLint [internal rule](https://github.com/eslint/eslint/blob/master/tools/internal-rules/no-useless-catch.js). Instead of just an internal rule, this should be a core rule.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[X] Suggests an alternate way of doing something (suggestion)
[ ] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
/* warn about useless try/catch statement */
try {
  throw new Error('whoops');
} catch (e) {
  throw e;
}

/* only warn about catch clause */
try {
  throw new Error('whoops');
} catch (e) {
  throw e;
} finally {
  console.log('done');
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**

As far as I know, there is no semantic difference between

```js
try {
  throw new Error('whoops');
} catch (e) {
  throw e;
}
```

and

```js
throw new Error('whoops');
```

Because of that, I can't think of a good reason why you would ever want a useless catch clause in any javascript codebase, seeing as how it just adds unnecessary clutter and complexity to your source code. It seems ESLint devs themselves agree since it exists as an internal rule.

**Are you willing to submit a pull request to implement this rule?**

If it is something that people want as a core rule, then sure.

**Comment by platinumazure** [comment]
I'll support this!

Although the rule definition exists, we would definitely need documentation and tests if we want to expose the rule as part of ESLint core.

**Comment by agrasley** [comment]
Cool. I'll look at existing rules and other PRs and try to come up with something.

**Comment by agrasley** [comment]
Okay, I've got a PR ready for this.

**Comment by ilyavolodin** [comment]
Sounds good to me.

**Comment by platinumazure** [comment]
@not-an-aardvark Thoughts on this? This was accepted a little early but I'm hoping you might be willing to support this. There's a PR linked as well. Thanks!
